os_is_not_exist_matcher.go

```diff
@@ -0,0 +1,32 @@
+package gomegamatchers
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/onsi/gomega/types"
+)
+
+func BeAnOsIsNotExistError() types.GomegaMatcher {
```

Comment:
That's only for matchers that need to hold on to some state.  If you look at matchers like `BeFalse`, they just implement the interface directly: https://github.com/onsi/gomega/blob/master/matchers/be_false_matcher.go